src/language-server/language-server.ts

```diff
@@ -64,20 +64,21 @@ export default class LanguageServer extends Handler {
         // communicated to us.
         const workspaceUri = getWorkspaceUri(params.rootUri, params.rootPath);
         if (!workspaceUri || workspaceUri.scheme !== 'file') {
-          reject(
+          const error = new Error(
               `Got invalid workspaceUri from client: ` +
               `${util.inspect(workspaceUri)}`);
-          return {capabilities: {}};
+          reject(error);
+          throw error;
```

Comment:
This is slightly tricky, but the call stack is separate from the Promise callback constructor because we're in the initialization callback. So to alert the client that initialization failed we need to throw, and to alert the promise that initialization failed we need to reject.